### PR TITLE
Install `ldid` with `brew`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Not sure if it's better to do this or just keep the previous code and `git checkout a23f0faadd29ec00a6b7fb2498c3d15af15a7100` to make sure we're running the code we want... until somebody manages to break SHA1DC. 😅

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
Discarded packaging alternatives:
* [Homebrew](https://formulae.brew.sh/formula/ldid) — `macOS is required for this software`
* [Source](git://git.saurik.com/ldid.git) — used by e.g. [pnpm](https://github.com/pnpm/pnpm/blob/3c2921647cdb76ac2e1ea739b1c73961f64b6c23/.github/workflows/release.yml#L25)

File: .github/workflows/test-deploy.yml [hunk 1]
```diff
@@ -105,9 +105,9 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
       - run: |
-          sudo apt install --yes libplist-dev
-          git clone --branch v2.1.5 git://git.saurik.com/ldid.git
-          sudo g++ -pipe -o /usr/bin/ldid ldid/ldid.cpp -I. -x c ldid/{lookup2.c,sha1.h} -lplist -lcrypto
+          echo "$CONDA/bin" >> "$GITHUB_PATH"
+          conda config --add channels conda-forge
+          conda install ldid
```
Comment (0x2b3bfa0) [inline .github/workflows/test-deploy.yml:110]:
```suggestion
          conda install ldid=2.1.2
```

**Comment by dacbd** [comment]
so just wait for https://github.com/conda-forge/ldid-feedstock/pull/7 to be safe then?

File: .github/workflows/test-deploy.yml [hunk 2]
```diff
@@ -105,9 +105,9 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
       - run: |
-          sudo apt install --yes libplist-dev
-          git clone --branch v2.1.5 git://git.saurik.com/ldid.git
-          sudo g++ -pipe -o /usr/bin/ldid ldid/ldid.cpp -I. -x c ldid/{lookup2.c,sha1.h} -lplist -lcrypto
+          echo "$CONDA/bin" >> "$GITHUB_PATH"
+          conda config --add channels conda-forge
+          conda install ldid=2.1.2
```
Comment (0x2b3bfa0) [inline .github/workflows/test-deploy.yml:110]:
```suggestion
          conda install ldid=2.1.5
```

**Comment by 0x2b3bfa0** [comment]
Blocked by https://github.com/conda-forge/ldid-feedstock/pull/7

**Comment by 0x2b3bfa0** [comment]
Alternatively, install from source and verify with thw following pearl from @dacbd:

```console
test "$(git show-ref v2.1.5)" == "31078501f34f2c672a41b2cda09494cd5fab65fe refs/tags/v2.1.5"
```

**Comment by dacbd** [comment]
> Alternatively, install from source and verify with thw following pearl from @dacbd:
> 
> ```
> test "$(git show-ref v2.1.5)" == "31078501f34f2c672a41b2cda09494cd5fab65fe refs/tags/v2.1.5"
> ```

:white_circle: thanks.... lol

**Review by dacbd — APPROVED** [review]
/rubber_stamp
#blocked

**Comment by 0x2b3bfa0** [comment]
Alternatively, https://github.com/Homebrew/homebrew-core/pull/112702

**Comment by 0x2b3bfa0** [comment]
@dacbd, https://github.com/Homebrew/homebrew-core/pull/112702 won the `brew` & `conda` race; should we use it?

**Comment by 0x2b3bfa0** [comment]
Note that `brew` requires a workaround (https://github.com/actions/runner-images/issues/6283) to run because GitHub was feeling trigger-happy and broke it.

```yaml
steps:
- run: for path in /home/linuxbrew/.linuxbrew/{s,}bin; do echo $path; done >> $GITHUB_PATH
- run: brew install ldid

**Comment by carlocab** [comment]
You can also use the `setup-homebrew` action, with something like
```yaml
    steps:
      - name: Set up Homebrew
        id: set-up-homebrew
        uses: Homebrew/actions/setup-homebrew@29b2973f18e5aca5443f4b3d87f68eda7a27b22b
```

File: .github/workflows/test-deploy.yml [hunk 3]
```diff
@@ -104,10 +104,8 @@ jobs:
       - run: npm ${{ inputs.release && 'publish' || 'publish --dry-run' }}
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - run: |
-          sudo apt install --yes libplist-dev
-          git clone --branch v2.1.5 git://git.saurik.com/ldid.git
-          sudo g++ -pipe -o /usr/bin/ldid ldid/ldid.cpp -I. -x c ldid/{lookup2.c,sha1.h} -lplist -lcrypto
+      - uses: Homebrew/actions/setup-homebrew@29b2973f18e5aca5443f4b3d87f68eda7a27b22b
```
Comment (0x2b3bfa0) [inline .github/workflows/test-deploy.yml:107]:
```suggestion
      # Step required "thanks" to https://github.com/actions/runner-images/issues/6283
      - uses: Homebrew/actions/setup-homebrew@29b2973f18e5aca5443f4b3d87f68eda7a27b22b
```

**Review by dacbd — APPROVED** [review]
:rubber_stamp:
but feels weird to use homebrew for not mac things... do we care, should we revisit this after a conda update?

**Comment by 0x2b3bfa0** [comment]
> _feels weird to use homebrew for not mac things_

It's indeed unusual, but “linuxbrew” is handy. I use it on my personal projects to install Hashicorp Packer on GitHub Actions with a single line. 🤷🏼‍♂️

**Comment by 0x2b3bfa0** [comment]
> _should we revisit this after a conda update?_

We could, if `conda` feels better for “not-scientific-Python things” 😈

**Comment by 0x2b3bfa0** [comment]
Thanks for all your help with https://github.com/Homebrew/homebrew-core/pull/112702, @carlocab!

**Pull request merged**